### PR TITLE
Moderation Queue

### PR DESCRIPTION
**Pull request by okbel**
# Moderation Queue

- ModerateQueue by Asset/Stream
	* Absolute Routes for Moderation
		* (`/admin/moderate/<asset_id>`)
		* Not found if the provided `<asset_id>` does not exist.
* All Streams: 
	* Streams link to each moderation Queue
* All refactored to use GraphQL
	* Query multiple Moderation Queues
	- Mutators
		- Ban User
		- Accept Comment  
		- Reject Comment   
- New mutations
	* setCommentStatus(commentId, status)
	* setUserStatus(userId, status)
* Refactor
	* Moderation Reducer
	* ActionButton
	* BanUserButton
	* BanUserDialog
	* Comment
         ... etc

Changes has been applied to both Front and Back end.
Building mutations for each action.

### CODE REVIEW

File: graph/mutators/comment.js [hunk 1]
```diff
@@ -162,22 +162,36 @@ const createPublicComment = (context, commentInput) => {
...
+    .then(res => res);
+};
+
 module.exports = (context) => {
 
   // TODO: refactor to something that'll return an error in the event an attempt
   // is made to mutate state while not logged in. There's got to be a better way
   // to do this.
-  if (context.user && context.user.can('mutation:createComment')) {
+  if (context.user && context.user.can('mutation:createComment', 'mutation:setUserStatus')) {
```
Comment (wyattjoh) [inline graph/mutators/comment.js]: This requires permission scheme breaks commenting for non-admin users. Also, shouldn't this be: `mutation:setCommentStatus`?

File: graph/mutators/user.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    return {
+      User: {
+        setUserStatus: (action) => setUserStatus(context, action)
+      }
+    };
+  }
+
+  return {
+    User: {
+      setUserStatus: () => {},
```
Comment (wyattjoh) [inline graph/mutators/user.js]: Should return `Promise.reject(errors.ErrNotAuthorized)`

**Review by wyattjoh — DISMISSED** [review]

**Comment by okbel** [comment]
@wyattjoh Changes added in the Review Meeting. Thanks!

**Review by gabelula — APPROVED** [review]
It looks good.

**Pull request merged**